docs/apps/lnw/es2k/readme.md

```diff
@@ -0,0 +1,263 @@
+
+# Linux Networking for ES2K
+
+Linux Networking involves incorporating comprehensive support for offloading various networking functions, such as L2 forwarding, L3 forwarding, ECMP, and VxLAN encapsulation and decapsulation intelligence to the IPU. This capability empowers overlay services to establish communication with endpoints through VxLAN tunnels, thereby extending the L2 segment across the underlay network. To achieve Linux networking support, we have utilized legacy OvS for overlay source MAC learning and VxLAN configurations, while relying on the kernel for underlay neighbor discovery, route management, and next-hop information.
+
+## Feature Overview
+
+This feature can run in two modes:
+
+- Slow path: All packets must land on the control plane and only control plane has the intelligence to forward the traffic.
+- Fast path: Only first unknown traffic (both SMAC and DMAC) will be sent to control plane and subsequent traffic should be forwarded by IPU E2100.
+
+To enable this feature we have,
+- `Infrap4d`: This process has p4runtime intelligence and acts as a gRPC server. Calls TDI front end to program IPU E2100.
+- `ovs-vswitchd`: This process is integrated with p4runtime intelligence and acts as a gRPC client. Programs IPU E2100 with control plane configuration and forwarding tables by communicating with gRPC server.
+- `p4rt-ctl`: This python CLI has p4runtime intelligence and acts as a gRPC client. Programs IPU E2100 with runtime rules by communicating with gRPC server.
+- `Kernel stack`: All underlay related configurations are picked by `kernel monitor` thread via netlink events in `infrap4d` and these are programmed in IPU E2100 by calling TDI front end calls.
+
+## Topology
+
+This topology breakup and configuration assumes all VMs are spawned on HOST VFs and control plane is running on ACC.
+
+![Linux Networking Topology](es2k-lnw-topology.png)
+
+### Topology breakup
+
+- Every VM spawned on top of a VF will have a corresponding port representor in ACC.
+- Every physical port will have a corresponding port representor in ACC.
+- Every physical port will have an uplink (APF netdev) in HOST and this uplink will have a corresponding port representor in ACC.
+- All port representors are associated with an OvS bridge.
+- For VxLAN egress traffic, underlay port should be associated with a termination bridge and IP to reach underlay network should be configured on top of this bridge.
+
+## Detailed Design
+
+### Slow path
+
+To enable slow path mode:
+- Start infrap4d process by disabling `kernel monitor` thread. Command: `infrap4d -disable-krnlmon`
+- Set environment variable `OVS_P4_OFFLOAD=false` before starting `ovs-vswitchd` process.
+
+In this mode, we need to associate VFs on top which VMs are created and its port representors and also physical ports with its port representors.
```

Comment:
```suggestion
In this mode, we need to associate VFs on top of which VMs are created and its port representors and also physical ports with its port representors.
```
This sentence is garbled. I can't figure out the relationships between VFs, VMs, physical ports, and port representors.

You're using _its_ (singular possessive) with plural nouns (_VFs_, _VMs_, _ports_, _representors_). The plural possessive is _their_. "ports with _their_ representors" (assuming that's the intended pairing).